diff --git a/tde-cmake/modules/tde_l10n_merge.pl b/tde-cmake/modules/tde_l10n_merge.pl
--- a/tde-cmake/modules/tde_l10n_merge.pl
+++ b/tde-cmake/modules/tde_l10n_merge.pl
@@ -1480,18 +1480,26 @@
     return @list;
 }
 
-sub quoted_translation
-{
+sub quoted_translation {
     my ($xml_mode, $lang, $string) = @_;
 
     $string = entity_decode($string) if $xml_mode;
-    $string =~ s/\\\"/\"/g;
+    $string =~ s/\\"/"/g;
+
+    # fallback: return original string if no translation hash for that language
+    unless (exists $translations{$lang}) {
+        return $string;
+    }
+
+    my $translation = $translations{$lang}{$string};
+
+    # fallback: return original string if no specific translation
+    $translation = $string if !defined $translation;
 
-    my $translation = $translations{$lang, $string};
-    $translation = $string if !$translation;
     $translation = entity_encode($translation) if $xml_mode;
-    $translation =~ s/\"/\\\"/g;
-    return $translation
+    $translation =~ s/"/\\"/g;
+
+    return $translation;
 }
 
 sub quoted_merge_translations
